src/clj/game/core/pick_counters.clj

```diff
@@ -76,11 +76,12 @@
 
 (defn pick-credit-providing-cards
   "Similar to pick-virus-counters-to-spend. Works on :recurring and normal credits."
-  ([provider-func outereid] (pick-credit-providing-cards provider-func outereid nil (hash-map) 0))
-  ([provider-func outereid target-count] (pick-credit-providing-cards provider-func outereid target-count (hash-map) 0))
-  ([provider-func outereid target-count selected-cards counter-count]
+  ([provider-func outereid] (pick-credit-providing-cards provider-func outereid nil (fn [_] 'false) 0 "" (hash-map) 0 0))
+  ([provider-func outereid target-count] (pick-credit-providing-cards provider-func outereid target-count (fn [_] 'false) 0 "" (hash-map) 0 0))
+  ([provider-func outereid target-count special? special-target special-label] (pick-credit-providing-cards provider-func outereid target-count special? special-target special-label (hash-map) 0 0))
+  ([provider-func outereid target-count special? special-target special-label selected-cards counter-count special-count]
```

Comment:
Oh for this, rather than put stealth specific stuff in pick-credit-providing-cards and just giving it an extra target argument, I kept it generic and made it take a predicate and label as well. So it's on the costs stuff to tell it what a stealth card is and what label it should be displaying to the user.